addons/resource/models/resource_calendar_attendance.py

```diff
@@ -70,23 +76,40 @@ def get_week_type(self, date):
 
     @api.depends('hour_from', 'hour_to')
     def _compute_duration_hours(self):
-        for attendance in self:
+        for attendance in self.filtered('hour_to'):
             attendance.duration_hours = (attendance.hour_to - attendance.hour_from) if attendance.day_period != 'lunch' else 0
 
-    @api.depends('day_period', 'duration_hours')
+    def _inverse_duration_hours(self):
+        if not self.calendar_id.duration_based:
```

Comment:
I think if you have two value in self you will have an error:
```suggestion
        if not any(attendance.calendar_id.duration_based for attendance in self) :
```